p2p/src/sync/mod.rs

```diff
@@ -201,15 +227,19 @@ where
     }
 
     /// Sends an event to the corresponding peer.
-    fn handle_peer_event(&mut self, event: SyncingEvent) {
+    fn handle_peer_event(&mut self, event: SyncingEvent) -> Result<()> {
         match event {
             SyncingEvent::Connected {
                 peer_id,
                 services,
                 sync_rx,
             } => self.register_peer(peer_id, services, sync_rx),
-            SyncingEvent::Disconnected { peer_id } => self.unregister_peer(peer_id),
+            SyncingEvent::Disconnected { peer_id } => {
+                self.request_tracker_peer_event_sender.send(PeerEvent::Disconnected(peer_id))?;
```

Comment:
Ideally, I didn't want the block sync manager to have to care about forwarding peer lifecycle events to the events tracker. An alternative would be if during request tracker creation it can subscribe to the peer events sent by backend.